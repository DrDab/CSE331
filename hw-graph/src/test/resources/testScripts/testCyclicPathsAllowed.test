# Create graphs w/ 2 and 3 nodes.
CreateGraph g2
AddNode g2 ALPHA
AddNode g2 BRAVO

CreateGraph g3
AddNode g3 ALPHA
AddNode g3 BRAVO
AddNode g3 CHARLIE

# Test cyclic paths are allowed on graph of 2 nodes.
AddEdge g2 ALPHA BRAVO ab
AddEdge g2 BRAVO ALPHA ba
ListChildren g2 ALPHA
ListChildren g2 BRAVO

# Test cyclic paths are allowed on graph of 3 nodes.
AddEdge g3 ALPHA BRAVO ab
AddEdge g3 BRAVO CHARLIE bc
AddEdge g3 CHARLIE ALPHA ca
ListChildren g3 ALPHA
ListChildren g3 BRAVO
ListChildren g3 CHARLIE