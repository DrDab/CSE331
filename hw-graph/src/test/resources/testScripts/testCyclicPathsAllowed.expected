# Create graphs w/ 2 and 3 nodes.
created graph g2
added node ALPHA to g2
added node BRAVO to g2

created graph g3
added node CHARLIE to g3
added node CHARLIE to g3
added node CHARLIE to g3

# Test cyclic paths are allowed on graph of 2 nodes.
added edge ab from ALPHA to BRAVO in g2
added edge ba from BRAVO to ALPHA in g2
the children of ALPHA in g2 are: BRAVO(ab)
the children of BRAVO in g2 are: ALPHA(ba)

# Test cyclic paths are allowed on graph of 3 nodes.
added edge ab from ALPHA to BRAVO in g3
added edge bc from BRAVO to CHARLIE in g3
added edge ca from CHARLIE to ALPHA in g3
the children of ALPHA in g3 are: BRAVO(ab)
the children of BRAVO in g3 are: CHARLIE(bc)
the children of CHARLIE in g3 are: ALPHA(ca)